ocaml/xapi/xapi_guest_agent.ml

```diff
@@ -205,7 +205,12 @@ let mutex = Mutex.create ()
 let get_initial_guest_metrics (lookup: string -> string option) (list: string -> string list) =
   let all_control = list "control" in
   let to_map kvpairs = List.concat (List.map (fun (xskey, mapkey) -> match lookup xskey with
-      | Some xsval -> [ mapkey, xsval ]
+      | Some xsval ->
+        if xskey = "control/feature-suspend" then
```

Comment:
This could be also guarded using `when` is a new match case above the existing one.

Reply:
I refactored this to use pattern matching.